test/server/echo.go

```diff
@@ -53,20 +56,40 @@ type handler struct {
 
 func (h handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	body := bytes.Buffer{}
+	r.ParseForm()
+
+	// If the request has form ?codes=int[,int]* return those codes, in sequence, rather than 200
+	err := setResponseFromCodes(r, w)
+	if err != nil {
+		body.WriteString("codes error: " + err.Error() + "\n")
+	}
+
 	body.WriteString("ServiceVersion=" + version + "\n")
 	body.WriteString("ServicePort=" + strconv.Itoa(h.port) + "\n")
 	body.WriteString("Method=" + r.Method + "\n")
 	body.WriteString("URL=" + r.URL.String() + "\n")
 	body.WriteString("Proto=" + r.Proto + "\n")
 	body.WriteString("RemoteAddr=" + r.RemoteAddr + "\n")
 	body.WriteString("Host=" + r.Host + "\n")
+
 	for name, headers := range r.Header {
 		for _, h := range headers {
 			body.WriteString(fmt.Sprintf("%v=%v\n", name, h))
 		}
 	}
+
+	hostname, err := os.Hostname()
+	if err == nil {
+		body.WriteString(fmt.Sprintf("Server instance hostname=%v\n", hostname))
+	}
+
+    for _, envExpr := range os.Environ() {
+    	if strings.HasPrefix(envExpr, "ECHO") {
+			body.WriteString(fmt.Sprintf("Server environment variable %v\n", envExpr))
+		}
```

Comment:
There is a flag called version that tries to do the same thing. Should we just use that one or there is some advantage?